### PR TITLE
Add percentile renderers

### PR DESCRIPTION
**Pull request by kyleli626**
Context: Adding percentile aggregations for measures.

Changes:
- Create new SqlPercentileExpression class that subclasses SqlFunctionExpression.
- Add engine-specific and default renderers visitor functions for new percentile expression.
- Add snapshot tests for new rendering.

### CODE REVIEW

File: metricflow/sql/render/expr_renderer.py
Comment (conglei) [inline metricflow/sql/render/expr_renderer.py:129]: Should we make this as the default implementation so it won't be duplicated among several sql engines?

File: metricflow/sql/sql_exprs.py
Comment (conglei) [inline metricflow/sql/sql_exprs.py:804]: Instead of passing two variables, we can creat a data class PercentileArgs and it has two fields.

File: metricflow/sql/render/expr_renderer.py
Comment (kyleli626) [reply metricflow/sql/render/expr_renderer.py:129]: yes - this is currently in the DefaultSqlExpressionRenderer, so it would be the default for snowflake, duckdb, postgres, and databricks.

Comment (conglei) [reply metricflow/sql/render/expr_renderer.py:129]: Oops, I was reading the code wrong 😂

File: metricflow/sql/sql_exprs.py
Comment (WilliamDee) [inline metricflow/sql/sql_exprs.py:825]: lets keep it consistent with the other and keep the parameters as a protected attribute in the class and expose it as a property for retrieval. I believe it also helps for type checking.

File: metricflow/sql/sql_exprs.py
Comment (WilliamDee) [inline metricflow/sql/sql_exprs.py:821]: `arg` seems kind of ambiguous. It goes into the `ORDER BY` clause so maybe rename to `order_by_arg`

**Review by WilliamDee — COMMENTED** [review]
this looks good!

File: metricflow/sql/sql_exprs.py
Comment (conglei) [inline metricflow/sql/sql_exprs.py:809]: 👍

**Review by conglei — APPROVED** [review]
It looks good to me! But want to get a second opinion from metricflow team :)

File: metricflow/sql/sql_exprs.py
Comment (tlento) [inline metricflow/sql/sql_exprs.py:815]: micro-nit: 
```suggestion
    """A percentile aggregation expression."""
```

**Review by tlento — APPROVED** [review]
Nice! I think you'll need to base this on latest main, there have been some updates that might affect this under the hood even if there aren't direct merge conflicts.

File: metricflow/sql/sql_exprs.py
Comment (kyleli626) [reply metricflow/sql/sql_exprs.py]: For sure! Will merge after the query_parser pr goes through. Think I neglected to rebase before merging today, my bad!

**Pull request merged**
